### PR TITLE
Check for account approval during checkin

### PR DESCRIPTION
**Pull request by quazi-irfan**

### CODE REVIEW

File: app/wama.web/Controllers/CheckInController.cs [hunk 1]
```diff
@@ -27,10 +29,15 @@ public IActionResult Index()
             if (ModelState.IsValid && !string.IsNullOrWhiteSpace(memberId))
             {
                 var loginCredential = await _CheckInService.GetLogInCredentialAsync(memberId);
-
-                if (loginCredential == null)
+                var userAccount = await _UserAccountService.GetUserAccountAsync(memberId);
+                
+                if(userAccount == null || loginCredential == null)
```
Comment (dubeme) [inline app/wama.web/Controllers/CheckInController.cs:34]: This condition is too broad, can you look into making it a bit more specific.

Comment (dubeme) [reply app/wama.web/Controllers/CheckInController.cs:34]: Here's my breakdown of the possible scenarios, hopefully it's helpful with breaking this down

loginCredential = lc
userAccount = ua


**lc == null** // No login credential

* Possible solutions
  * If **ua == null** (see below for how to deal with it)
  * Else
    * Most likely, the account hasn't been verified - (Talk to the front desk folks)
    * Something catastrophic has happened - (Don't worry about this)

**ua == nul**l // No user account
* No account with the member ID exists, create new account

**Review by dubeme — CHANGES_REQUESTED** [review]
See comments

File: app/wama.web/Controllers/CheckInController.cs (hunk 1 above)
Comment (quazi-irfan) [reply app/wama.web/Controllers/CheckInController.cs:34]: How about the following?

```
if ua == null
    Message(ID_Does_not_exits)
else {
    if( !userAccount.HasBeenApproved)
        Message(Account_Approval_Pending)
    else 
        checkin_sucessful_page
}
```

Comment (dubeme) [reply app/wama.web/Controllers/CheckInController.cs:34]: Looking good

**Review by dubeme — CHANGES_REQUESTED** [review]

**Pull request closed without merging**